From 9a30058ab2e4c82cb5a930cb86b4668b68c593b2 Mon Sep 17 00:00:00 2001
From: Alexey Puzhevich <alexey.puzhevich@intel.com>
Date: Mon, 19 May 2025 15:36:05 +0300
Subject: [PATCH] [SW-204735] packages: zephyr: target for zephyr size report

This patch adds additional targets  for  Zephyr modules to produce dumps
to view and analyse RAM  and  ROM usage in generated images. It uses the
final image  and  give information about  size of symbols and code being
used in both RAM and ROM. New targets are:
  <modulename>-size-report-ram
  <modulename>-size-report-rom
  <modulename>-size-report

Change-Id: Ic464b3444536838883d7f90389ed49e25ebc6eac
Signed-off-by: Alexey Puzhevich <alexey.puzhevich@intel.com>
---

diff --git a/package/zephyr/zephyr.mk b/package/zephyr/zephyr.mk
index 8584d9e..da2aab7 100644
--- a/package/zephyr/zephyr.mk
+++ b/package/zephyr/zephyr.mk
@@ -195,6 +195,27 @@
 	$(foreach component,$(ZEPHYR_COMPONENTS_LIST),$(call ZEPHYR_COMPONENT_VERSION_HOOK,${component}))
 endef
 
+define ZEPHYR_SIZE_REPORT
+	echo "Zephyr $(1) size report" && \
+	cd $(2) && \
+	python3 zephyr/scripts/footprint/size_report \
+		-k build/zephyr/zephyr.elf \
+		-z zephyr/ \
+		-w zephyr \
+		-o build/zephyr/ \
+		-d 99 \
+		$(1)
+endef
+
+%-size-report-ram: %-build
+	@$(call ZEPHYR_SIZE_REPORT,ram,$(BUILD_DIR)/$*-custom)
+
+%-size-report-rom: %-build
+	@$(call ZEPHYR_SIZE_REPORT,rom,$(BUILD_DIR)/$*-custom)
+
+%-size-report: %-build
+	@$(call ZEPHYR_SIZE_REPORT,all,$(BUILD_DIR)/$*-custom)
+
 ZEPHYR_DEPENDENCIES += $(HL_ZEPHYR_PACKAGES)
 
 zephyr-configure: $(ZEPHYR_DEPENDENCIES)
